CefSharp.OffScreen/ChromiumWebBrowser.cs

```diff
@@ -614,20 +678,77 @@ void IRenderWebBrowser.InvokeRenderAsync(BitmapInfo bitmapInfo)
         /// in it's lock scope.
         /// </summary>
         /// <param name="bitmapInfo">information about the bitmap to be rendered</param>
+        /// <exception cref="InvalidOperationException">When background image is null when on blend mode.</exception>
         public virtual void InvokeRenderAsync(BitmapInfo bitmapInfo)
         {
             var gdiBitmapInfo = (GdiBitmapInfo)bitmapInfo;
-            if (bitmapInfo.CreateNewBitmap)
+
+            if (BlendingConfiguration == PopupBlending.Separate)
             {
-                if (Bitmap != null)
+                if (bitmapInfo.CreateNewBitmap)
                 {
-                    Bitmap.Dispose();
-                    Bitmap = null;
+                    if (Bitmap != null)
+                    {
+                        Bitmap.Dispose();
+                        Bitmap = null;
+                    }
+
+                    Bitmap = gdiBitmapInfo.CreateBitmap();
                 }
+            }
+            else if (BlendingConfiguration == PopupBlending.Blend)
```

Comment:
This would be more effective, but I think the one bitmap we should save still is the background, since in case of popup, "Bitmap" is set to the popup image only. That would probably allow us to keep only one image, and as you said only merge when really needed. I'll try it right now.